syntax = "proto3";

package protos;

enum JournalScoreProvider {
  UNKNOWN_JOURNAL_SCORE_PROVIDER = 0;
  COMPUTED = 1;
  SCI_SCORE = 2;
  SCIMAGO = 3;
}

message ComputedScore {
  // Percentile rank of this score across all scores for all years
  float percentile_rank = 1;
}

message SciScore {
  string raw_data_url = 1;

  float avg_score = 2;

  // Percentile rank of this score across all scores for all years
  float percentile_rank = 3;
}

message Scimago {
    string raw_data_url = 1;

    // Valid values are in the range [1, 4], where 1 is the highest.
    // Missing values are represented as 0.
    int32 best_quartile = 2;
}

message JournalScoreMetadata {
  oneof provider_metadata {
    ComputedScore computed_score = 4;
    SciScore sci_score = 5;
    Scimago scimago = 6;
  }
}

message JournalScore {
  int64 id = 1;

  int64 journal_id = 2;

  int32 year = 3;

  JournalScoreProvider provider = 4;

  JournalScoreMetadata metadata = 5;

  string created_by = 6;

  int64 created_at_usec = 7;
}

message Journal {
  // DB generated sequential ID
  int64 id = 1;

  // Required: Name of the journal
  string name = 2;

  // Optional: Print ISSN if available
  string print_issn = 3;

  // Optional: Electronic ISSN if available
  string electronic_issn = 4;
}
